api/circle/landing/[token].ts

```diff
@@ -0,0 +1,62 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { adminClient } from '../../../api-lib/gql/adminClient';
+import { errorResponse } from '../../../api-lib/HttpError';
+import { Awaited } from '../../../api-lib/ts4.5shim';
+
+export default async function handler(req: VercelRequest, res: VercelResponse) {
+  // TODO: does this have some kind of CORS protection?
+  try {
+    let token: string | undefined;
```

Comment:
probably can simplify it this way
```ts
const token = (() => {
      if (typeof req.query.token == 'string') {
        return req.query.token;
      } else if(req.query.token instanceof Array && req.query.token[0]) {
         return req.query.token[0];
      } else {
        throw new Error('empty token provided');
      }
    })();
```

Reply:
to me this is less simple haha, idk!

Reply:
i simplified it slightly, i think its ok